### ISSUE TITLE
Inconsistent People Results for Different Users

### ISSUE THREAD

**Issue by Dave-Buck** [microsoft-search/pnp-modern-search#1570]
### Check existing issues before logging new ones

**Version used**
Ex: 4.4.1

**Describe the bug**
Installing the search Webpart and searching against LocalPeopleResults (SourceID: b09a7990-05ea-4af9-81ef-edfab16c4e31) returns different results depending on the signed in user for certain queries (As far as I am aware, people results are not subjected to permissions / audience targeting?).

I first noticed this on my own Webpart which utilises the search API, and was able to reproduce in multiple tenancies, as well as the SharePoint Search Query Tool, and this Webpart. This may suggest that there is an issue with the search API, but I have had no luck with Microsoft Support, and know the PnP team work closely/for Microsoft.

This issue has arisen in the last week or so, clients are now no longer able to use the module as intended.

**To Reproduce**
1. Create Demo Tenancy (or use existing)
2. Install Webpart (accept API permissions etc)
3. Configure the Webpart in the following manner (All other config left as default):
a. Result source ID: LocalPeopleResults
b. Selected properties: AuthorOWSUSER, JobTitle, UserName, WorkPhone.
c. Layout: People
d. Static Query: LastName:*
4. When signed in with Global Admin for demo tenancy, you should see 13 results, when signed in with standard user, you will get none.

Notes: I have assigned the standard user the same licenses, and then applied (from what I can see) the same permissions to the user as the Global Admin, and I am still getting inconsistent results.

**Expected behaviour**
For Local People Results, I would expect that irrespective of the user, the same results are returned for the same API request. As mentioned, unless you know, I do not believe People results are subjected to permissions / audience targetting.

**Screenshots**
Here is the request that is sent for this module:
![image](https://user-images.githubusercontent.com/33024481/143910400-aa7b33f2-df7e-4fd1-9cad-595854625bb6.png)

I am able to send other request payloads for my other module / SharePoint Query Tool if required.

**Desktop (please complete the following information):**
 - Browser Chrome

**Additional context**
I am happy to give access to my demo tenancy should you need quick access to a broken example.

If this is the wrong place for this issue due to it looking like an API issue, can you please point me to an open ticket / user voice relating to this? As I am not able to find anything as of yet.

Please do let me know if you require any additional information.

**Comment by FranckyC** [comment]
Hi @Dave-Buck, more a question for @wobba. I can't really tell what happens here :/

**Comment by rajeshb4** [comment]
Hi @Dave-Buck, I also noticed this happening in our tenant. Not sure what changed all of a sudden. Opened a case with Microsoft. I'll update you if I hear back from MS.

**Comment by Dave-Buck** [comment]
Hi @rajeshb4, thanks for the update. I have also raised an MS support ticket, I have a meeting with a Support Engineer from SharePoint Online Dev team under premium support, so will hopefully will find more information soon! I will keep you posted.

**Comment by wobba** [comment]
You should see the same number of results independent of the user when querying for people. And the API permissions are only required if you use the Graph Toolkit display or the Graph search provider.

`AuthorOWSUSER` is not applicable to user profiles - it's a property for items. `PreferredName` is a better one for the profile's name.

**Comment by Dave-Buck** [comment]
Hi @wobba, thanks for the response.

I understand, if I remove the AuthorOWSUSER property, the issue still remains. And thank you for clarifying that no permissions should be at play here.

It appears there is something strange going on with the SharePoint search API, across multiple tenancies. Are you happy for me to to leave this ticket open until I get a response from Microsoft?

**Comment by wobba** [comment]
Sure, and feel free to update here with information you get.

**Comment by Dave-Buck** [comment]
I have came back from the weekend and found that I appear to no longer be experiencing the same issue. I try with different accounts, and I am now getting the expected results for each user. @rajeshb4  is this the same for you?

I have noticed however, that wildcards are no longer working as expected. If I set my KQL so something like:
PreferredName:* 
Or
JobTitle:*

I get no results. But if I have the KQL, PreferredName:a*, I do get results. Anyone else able to replicate?
